packages/xo-web/src/xo-app/xoa/licenses/index.js

```diff
@@ -100,45 +139,70 @@ export default class Licenses extends Component {
     this.componentDidMount = this._updateLicenses
   }
 
-  _updateLicenses = () =>
-    Promise.all([getLicenses('xosan'), getLicenses('xosan.trial')])
-      .then(([xosanLicenses, xosanTrialLicenses]) => {
+  _updateLicenses = () => {
+    this.setState({ licenseError: undefined })
+
+    return getLicenses()
+      .then(licenses => {
+        const { xoa, xosan } = groupBy(licenses, license =>
+          license.productTypes.includes('xo')
+            ? 'xoa'
+            : license.productTypes.includes('xosan')
+            ? 'xosan'
+            : 'other'
+        )
         this.setState({
-          xosanLicenses,
-          xosanTrialLicenses,
-          licenseError: undefined,
+          licenses: {
+            xoa,
+            xosan,
+          },
         })
       })
       .catch(error => {
         this.setState({ licenseError: error })
       })
+  }
 
   _getProducts = createSelector(
-    () => this.state.xosanLicenses,
-    () => this.props.xosanSrs,
-    (xosanLicenses, xosanSrs) => {
-      const products = []
-      if (get(() => xosanLicenses.state) === 'register-needed') {
+    () => this.state.licenses,
+    licenses => {
+      if (get(() => licenses.xosan.state) === 'register-needed') {
         // Should not happen
         return
       }
 
-      // XOSAN
-      const boundSrs = []
-      forEach(xosanLicenses, license => {
-        if (license.boundObjectId !== undefined) {
-          boundSrs.push(license.boundObjectId)
+      const now = Date.now()
+      const products = []
+
+      // --- XOSAN ---
+      forEach(licenses.xosan, license => {
+        if (!(license.expires < now) && license.productId === 'xosan') {
```

Comment:
A comment which explains that the license can rarely be expires will be helpful.